src/server/system_services/bucket_server.js

```diff
@@ -151,6 +153,36 @@ async function create_bucket(req) {
             );
             changes.insert.tieringpolicies = [tiering_policy];
             changes.insert.tiers = [tier];
+
+            // Attach a `GLACIER` tier to the bucket if it is not namespace.caching
+            // and appropriate configuration is set
+            if (!req.rpc_params.namespace?.namespace?.caching) {
+                if (config.TMFS_BUCKET_TIER2_AUTOCONF && config.BLOCK_STORE_FS_TMFS_ENABLED) {
+                    const tier2_name = `${bucket_with_suffix}_auto_tier2`;
+
+                    const tier2_mirrors = [{
+                        _id: system_store.new_system_store_id(),
+                        spread_pools: [default_pool._id]
+                    }];
+
+                    tier_server.check_tier_exists(req, tier2_name);
+                    const tier2 = tier_server.new_tier_defaults(
+                        tier2_name,
+                        req.system._id,
+                        chunk_config._id,
+                        tier2_mirrors,
+                        STORAGE_CLASS_GLACIER,
+                    );
+
+                    changes.insert.tiers.push(tier2);
```

Comment:
There's also first.bucket that gets created in system_server create_system.
Would be good to share this code flow for both cases.

Reply:
Moved this into a common function. Doesn't handles any customizations yet but does extend the functionality to create_system.

Tested: `npm run web` => `first.bucket` had 2 tiers attached to it.